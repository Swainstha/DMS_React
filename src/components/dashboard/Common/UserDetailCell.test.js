import React from 'react';
import UserDetailCell from './UserDetailCell';

import {
    configure,
    shallow,
} from 'enzyme';
import EnzymeAdapter from 'enzyme-adapter-react-16';
// JestHook.mock('expo-font');
configure({
    adapter: new EnzymeAdapter
})


const dummy_data={data:
    {   id:1, 
        name:"john",
        detail:"health assistant",
        about:"lives in new jersey", 
        image:"https://x.com/23/350/420", 
        color:"orange"}}
const setup = (props = {}) => {
    return shallow( < UserDetailCell
     {...props}  />
    )
}

const findByTestAttr=(wrapper, val)=>{
    return wrapper.find(`[data-test='${val}']`)
}

 it("snapshot test UserDetailCell", () => {
        let wrapper = setup(dummy_data);
        expect(wrapper).toMatchSnapshot();
}) 
it("UserDetailCell props",()=>{

    let wrapper = setup(dummy_data);
    const avatar=findByTestAttr(wrapper, `avatar-component`);
    const h5= wrapper.find('h5')
    const p = wrapper.find('p')
    const span =wrapper.find('span')
    expect(avatar).toHaveLength(1);
    expect(p).toHaveLength(1);
    expect(span).toHaveLength(1);
    expect(h5).toHaveLength(1);
    expect(avatar.props().src).toEqual("https://x.com/23/350/420");
    expect(span.props().children).toEqual("health assistant");
    expect(p.props().children).toEqual("lives in new jersey");
    expect(h5.props().children).toEqual("john");







})

