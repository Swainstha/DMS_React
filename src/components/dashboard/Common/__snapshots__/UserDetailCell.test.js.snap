// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test UserDetailCell 1`] = `ShallowWrapper {}`;
